### ISSUE TITLE
The multiple solutions

### ISSUE THREAD

**Issue by BigbigVegetable** [USTC-JialunPeng/Diverse-Structure-Inpainting#12]
How are the multiple solutions mentioned in your article reflected in the training and testing process?

**Comment by USTC-JialunPeng** [comment]
As mentioned in our supplementary material, the diversity is fully determined by the learned conditional distribution for structure generation (since the texture generation has no randomness). 

The corresponding training process is the cross entropy (or NLL loss) between the output discrete distribution of PixelCNN and structural encoding indices of ground truth. 
https://github.com/USTC-JialunPeng/Diverse-Structure-Inpainting/blob/0def26df30f291cdfd68c248228366f9d0a8f7e5/train_structure_generator.py#L225

The corresponding testing process is the iteratively sampling from the output discrete distribution of PixelCNN for the generation of discrete structure features. 
https://github.com/USTC-JialunPeng/Diverse-Structure-Inpainting/blob/0def26df30f291cdfd68c248228366f9d0a8f7e5/test.py#L167-L178
